VMRecovery/ResourceManager/Common-Functions.psm1

```diff
@@ -755,6 +763,92 @@ function write-log
     }        
 }
 
+Function Build-PostData
+{
+    Param(
+		[Parameter(Mandatory=$true)]
+		[string]$EventName,
+        [Parameter(Mandatory=$false)]
+        [string]$Scriptname,
+        [Parameter(Mandatory=$false)]
+        [string]$Command,
+        [Parameter(Mandatory=$false)]
+        [string]$Scriptversion,
+        [Parameter(Mandatory=$false)]
+        [string]$Message,
+        [Parameter(Mandatory=$false)]
+        [string]$Duration,
+        [Parameter(Mandatory=$true)]
+        [string]$RunId
+	)
+    $InstrumentKey = "7d48ea58-f6fe-4795-844b-ea580b90be26"
+    if (RanfromCloudshell){$Environment = "CloudShell"} else {$Environment = "Powershell"}
+    $CustomProperties = @{
+	"Script Name" = "$scriptname";
+	"DeveloperMode" = "false"
+    "Time" = [Datetime]::UtcNow.ToString("yyyy-MM-dd HH:mm:ss")
+    "Message" = $Message
+    "Command" = $Command
+    "ScriptVersion" = $Scriptversion
+    "Environment" = $Environment
+    "Duration" = $Duration
+    "RunID" = $RunId
+    }
+
+	Try {
+		Return @{
+			name = "Microsoft.ApplicationInsights.Dev.$InstrumentKey.Event";
+			time = [Datetime]::UtcNow.ToString("yyyy-MM-dd HH:mm:ss");
+			iKey = $InstrumentKey;
+			data = @{
+				baseType = "EventData";
+				baseData = @{
+					name = $eventName;
+					properties = $CustomProperties;
+				}
+			};
+		}
+	} Catch {
+		Throw $_
+	}
+}
+
+
+Function LogToAppInsight
+{
+    Param(
+		[Parameter(Mandatory=$true)]
+		[string]$EventName,
+        [Parameter(Mandatory=$false)]
+        [string]$Scriptname,
+        [Parameter(Mandatory=$false)]
+        [string]$Command,
+        [Parameter(Mandatory=$false)]
+        [string]$Scriptversion="1.0.0",
+        [Parameter(Mandatory=$false)]
+        [string]$Message="None",
+        [Parameter(Mandatory=$false)]
+        [string]$Duration,
+        [string]$RunID
+	)
+
+    Try {
+        
+        $postData = Build-PostData -EventName $EventName  -Scriptname $Scriptname -Command $Command -Scriptversion $Scriptversion -Message $Message -Duration $Duration -RunId $RunID| ConvertTo-Json -Depth 5   
+	    Try {
+            write-log "[Running] Posting Telemetry data to AppInsights" 
+		    $Response = Invoke-RestMethod -Method POST -Uri "https://dc.services.visualstudio.com/v2/track" -ContentType "application/json" -Body $postData

```

Comment:
NIT: Indention.

Also, I am guessing you had 3 log lines when debugging the AppInsights addition.  You might consider if all three are necessary when rolling to PROD usage... given you _should_ now have some telemetry.

Reply:
Good point removed the extra one with the -LogOnly option. The other two are more for the user to see its progress.